### ISSUE TITLE
Focusing a webview does not cause onDidChangeActiveTextEditor to be fired

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/vscode#59139]
Steps to Reproduce:

1. In an extension that registers `onDidChangeActiveTextEditor`
2. Open the markdown preview to the side of a markdown file
3. Click back and forth between the preview and the editor

**Bug**
Notice that `onDidChangeActiveTextEditor` is not fired. It should be fired with `undefined` when switching to the preview and then again with the editor when switching back to it

**Comment by mjbvz** [comment]
I'm not sure but I think this bug may have been reveled by https://github.com/Microsoft/vscode/pull/58887 Specifically `_updateState` is now no longer fired when switching from an editor to a webview with the output panel hidden

**Comment by bpasero** [comment]
This is a consequence of no longer listening to the editor blur event (which I still think is the right thing to do). 

However I think it is fair to say that the active editor should be set to `undefined` if the workbench active editor is not a text editor (or normal editor or diff editor).

@jrieken I believe the `MainThreadDocumentAndEditorStateComputer` needs to also account for `IEditorService#onActiveEditorChange()` and then react only iff the active editor is either null (e.g. when for an empty editor group) or when a non-text editor gets opened.

**Comment by bpasero** [comment]
Should be OK, however given the fact that editor groups can be empty there is a chance that opening an editor to the side of the current one first fires an event first with an empty editor and then with the real editor.

I used to have timeouts and buffers in the old world that resulted in weird behavior, so I decided to remove them for the grid world and get the true events. If this is an issue in a particular case, I suggest to do the buffering at the location where this is an issue.
